File: archive/sensor-readings-uploader-android-app/sensor-uploader/src/main/java/altermarkive/uploader/Log.java
package altermarkive.uploader;

import java.util.Date;

public class Log {
    public static String LOG_FILE = "runtime.log";

    public static void log(String priority, String tag, String entry) {
        entry = String.format("%s\n%s: %s\n%s\n\n", new Date().toString(), priority, tag, entry);
        Storage.appendText(LOG_FILE, entry);
    }

    @SuppressWarnings("unused")
    public static void v(String tag, String entry) {
        //android.util.Log.v(tag, entry);
        log("VERBOSE", tag, entry);
    }

    @SuppressWarnings("unused")
    public static void d(String tag, String entry) {
        //android.util.Log.d(tag, entry);
        log("DEBUG", tag, entry);
    }

    @SuppressWarnings("unused")
    public static void i(String tag, String entry) {
        //android.util.Log.i(tag, entry);
        log("INFO", tag, entry);
    }

    @SuppressWarnings("unused")
    public static void w(String tag, String entry) {
        //android.util.Log.w(tag, entry);
        log("WARN", tag, entry);
    }

    @SuppressWarnings("unused")
    public static void e(String tag, String entry) {
        //android.util.Log.e(tag, entry);
        log("ERROR", tag, entry);
    }

    public static String getStackTraceString(Throwable throwable) {
        return android.util.Log.getStackTraceString(throwable);
    }
}

File: fall-detector/src/main/java/altermarkive/guardian/Log.kt
package altermarkive.guardian

object Log {
    fun println(level: Int, tag: String, entry: String) {
        android.util.Log.println(level, tag, entry)
        Sampler.instance?.data()?.log(level, tag, entry)
    }

    @Suppress("unused")
    fun v(tag: String, entry: String) {
        android.util.Log.v(tag, entry)
        Sampler.instance?.data()?.log(android.util.Log.VERBOSE, tag, entry)
    }

    @Suppress("unused")
    fun d(tag: String, entry: String) {
        android.util.Log.d(tag, entry)
        Sampler.instance?.data()?.log(android.util.Log.DEBUG, tag, entry)
    }

    @Suppress("unused")
    fun i(tag: String, entry: String) {
        android.util.Log.i(tag, entry)
        Sampler.instance?.data()?.log(android.util.Log.INFO, tag, entry)
    }

    @Suppress("unused")
    fun w(tag: String, entry: String) {
        android.util.Log.w(tag, entry)
        Sampler.instance?.data()?.log(android.util.Log.WARN, tag, entry)
    }

    @Suppress("unused")
    fun e(tag: String, entry: String) {
        android.util.Log.e(tag, entry)
        Sampler.instance?.data()?.log(android.util.Log.ERROR, tag, entry)
    }
}